functorch/functorch/_src/partitioners.py

```diff
@@ -164,6 +174,100 @@ def default_partition(
+# after calling the compiled forward graph,
+# and running "input_mutation_submodule".
+def move_input_mutations_into_submodule(fx_g: fx.GraphModule) -> List[bool]:
+    mutable_nodes: List[Node] = []  # the copy_() and as_strided_() nodes, that we'll delete move to a submodule.
+    mutated_inputs: List[Node] = []
+    node_map: Dict[Node, Node] = {}
+    node_to_placeholder: Dict[Node, Node] = OrderedDict()  # mapping of nodes from old graph to placeholder in new submodule.
+    mutation_subgraph = fx.Graph()
+    model_input_to_mutation: Dict[Node: bool] = OrderedDict()
+
+    # First, default all model inputs to not having mutations
+    for n in fx_g.graph.nodes:
+        if n.op == 'placeholder':
+            model_input_to_mutation[n] = False
+
+    def remap_input(x):
+        if x in node_map:
+            return node_map[x]
+        if x not in node_to_placeholder:
+            # Add the copy_() mutated input argument as a submodule input.
+            placeholder_node = mutation_subgraph.placeholder(x.name, type_expr=x.type)
+            placeholder_node.meta = copy.copy(x.meta)
+            node_to_placeholder[x] = placeholder_node
+        return node_to_placeholder[x]
+
+    inpt_nodes = [n for n in fx_g.graph.nodes if n.op == 'placeholder']
+    for n in fx_g.graph.nodes:
+        if not isinstance(n.target, torch._ops.OpOverload):
+            continue
+        mutable_schema_args = [
+            a for a in n.target._schema.arguments
+            if a.alias_info is not None and a.alias_info.is_write
+        ]
+        if len(mutable_schema_args) == 0:
+            continue
+        # First some basic assertions.
+        # This graph should be entirely functional, but might contain
+        # some copy_() mutation nodes, that copy data back into program inputs.
+        # Any mutable ops that we find should correspond to copy_() nodes.
+        assert n.target == torch.ops.aten.copy_.default or \
+               n.target == torch.ops.aten.as_strided_.default, \
+               "Given a non-functionalized graph."
+        # Add the copy_() node to the submodule, remember which nodes are mutated input
+        # so we can return them in the original graph.
+        mutable_nodes.append(n)
+        if n.target == torch.ops.aten.copy_.default:
+            original_input = n.args[0]
+            mutated_input = n.args[1]
+            if mutated_input not in mutated_inputs:
+                mutated_inputs.append(mutated_input)
+            # Figure out which inputs were mutated
+            model_input_to_mutation[original_input] = True
+        new_node = mutation_subgraph.node_copy(n, remap_input)
+        node_map[n] = new_node
+    # Add the submodule to the end of the original graph
+    mutation_subgraph.lint()
+    mutation_submodule = lift_subgraph_as_module(fx_g, mutation_subgraph)
+    submodule_name = 'input_mutation_submodule'
+    fx_g.add_submodule(submodule_name, mutation_submodule)
```

Comment:
Given the code in AOTAutograd, it doesn't seem like you make use of the fact that the mutation module is a submodule at all, since you don't have a `call_submodule` call anymore. In that case, might as well just not add it as a submodule and just return it from this function. Partitioning in a true sense.